dask/dataframe/io/parquet/fastparquet.py

```diff
@@ -288,154 +279,408 @@ def read_metadata(
                 meta[catcol] = meta[catcol].cat.set_categories(pf.cats[catcol])
             elif meta.index.name == catcol:
                 meta.index = meta.index.set_categories(pf.cats[catcol])
-        if gather_statistics and pf.row_groups:
-            stats = []
-            if filters is None:
-                filters = []
-
-            skip_cols = set()  # Columns with min/max = None detected
-            # make statistics conform in layout
-            for (i, row_group) in enumerate(pf.row_groups):
-                s = {"num-rows": row_group.num_rows, "columns": []}
-                for i_col, col in enumerate(pf.columns):
-                    if col not in skip_cols:
-                        d = {"name": col}
-                        cs_min = None
-                        cs_max = None
-                        if pf.statistics["min"][col][0] is not None:
-                            cs_min = pf.statistics["min"][col][i]
-                            cs_max = pf.statistics["max"][col][i]
-                        elif (
-                            dtypes[col] == "object"
-                            and row_group.columns[i_col].meta_data.statistics
-                        ):
-                            cs_min = row_group.columns[
-                                i_col
-                            ].meta_data.statistics.min_value
-                            cs_max = row_group.columns[
-                                i_col
-                            ].meta_data.statistics.max_value
-                            if isinstance(cs_min, (bytes, bytearray)):
-                                cs_min = cs_min.decode("utf-8")
-                                cs_max = cs_max.decode("utf-8")
-                        if None in [cs_min, cs_max] and i == 0:
-                            skip_cols.add(col)
-                            continue
-                        if isinstance(cs_min, np.datetime64):
-                            tz = getattr(dtypes[col], "tz", None)
-                            cs_min = pd.Timestamp(cs_min, tz=tz)
-                            cs_max = pd.Timestamp(cs_max, tz=tz)
-                        d.update(
-                            {
-                                "min": cs_min,
-                                "max": cs_max,
-                                "null_count": pf.statistics["null_count"][col][i],
-                            }
-                        )
-                        s["columns"].append(d)
-                # Need this to filter out partitioned-on categorical columns
-                s["filter"] = fastparquet.api.filter_out_cats(row_group, filters)
-                s["total_byte_size"] = row_group.total_byte_size
-                s["file_path_0"] = row_group.columns[0].file_path  # 0th column only
-                stats.append(s)
```

Comment:
The `FastParquetEngine` has been completely rewritten to align with the "pyarrow" engines.  The logic is not new, but was previously used for pyarrrow only.